### PR TITLE
Add audit trail logging to boefje crud actions in boefje

### LINKED ISSUE #3603: Add audit events to Boefje creation  / deletion.
### Detailed description

As requested we need to provide Audit events when certain actions take place. This includes creating/deleting/changing boefjes and their variants.
This also includes setting or changing boefje configurables.

### Feature benefit/User story

Our installation admins want to keep track of changes by their users.

### Specifications

https://github.com/minvws/nl-kat-coordination/issues/3043

### PR DESCRIPTION
**Pull request by madelondohmen**
### Changes

This PR adds logging when a boefje (variant) is being created or edited.

### Issue link
Closes #3603 

### Demo
Creating Boefje:
afbeelding

Editing Boefje:
afbeelding

### QA notes
- Create a new Boefje, check the rocky logs
- Create a new Boefje variant, check the rocky logs
- Edit an existing Boefje, check the rocky logs

---

### Code Checklist

- [x] All the commits in this PR are properly PGP-signed and verified.
- [x] This PR only contains functionality relevant to the issue.
- [ ] I have written unit tests for the changes or fixes I made.
- [x] I have checked the documentation and made changes where necessary.
- [x] I have performed a self-review of my code and refactored it to the best of my abilities.

- [ ] Tickets have been created for newly discovered issues.
- [ ] For any non-trivial functionality, I have added integration and/or end-to-end tests.
- [ ] I have informed others of any required `.env` changes files if required and changed the `.env-dist` accordingly.
- [ ] I have included comments in the code to elaborate on what is not self-evident from the code itself, including references to issues and discussions online, or implicit behavior of an interface.

---

## Checklist for code reviewers:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---

## Checklist for QA:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.

### CODE REVIEW

File: rocky/katalogus/views/boefje_setup.py [hunk 1]
```diff
@@ -44,6 +47,7 @@ def form_valid(self, form):
         plugin = create_boefje_with_form_data(form_data, self.plugin_id, self.created)
 
         try:
+            logger.info("Creating boefje", event_code=800025, boefje=plugin)
```
Comment (ammar92) [inline rocky/katalogus/views/boefje_setup.py:50]: Perhaps only log the plugin ID or name instead of the complete structure? Like you do at other places

Comment (underdarknl) [reply rocky/katalogus/views/boefje_setup.py:50]: I'm in favor of  sending the whole boefje to audit logs.

**Comment by underdarknl** [comment]
We should probably discuss if we want to log these things in the views, or in the Katalogus_client (and other clients).
Doing to in the views creates the risk that we miss one or more. Doing so in the objects their create/update/delete methods probably makes it harder to log the correct context, but does harden against missed logging implementations.

File: rocky/katalogus/views/boefje_setup.py (hunk 1 above)
Comment (dekkers) [reply rocky/katalogus/views/boefje_setup.py:50]: This goes against the structured logging we try to do. You get a log message like:

```[info] Creating boefje                boefje=Boefje(id='5f569b52-7140-4c31-82e9-c3c09d73fad9', name='test5', version=None, authors=None, create
d='2024-10-03 14:15:47.319434', description='test', related=[], enabled=False, type='boefje', scan_level=SCAN_LEVEL.L0, consumes=set(), produces=set(), options=None, runnable_hash=None, interval=None, boefje_schema=None, oci_image='test', oci_arguments=[]) event_code=800025 ip=172.30.0.1 organization_code=test request_id=dcfb0bbc-5858-425c-9ceb-3be8121bd430 user_id=1```

Other than that it is ugly and logs a lot of uninteresting data, how are you supposed to easily filter all logs messages for a boefje with the name 'test5'? The whole idea of structured logging is that you can have filter with for example a key boefje_name and value 'test5'...

**Comment by stephanie0x00** [comment]
## Checklist for QA:

- [x] I have checked out this branch, and successfully ran a fresh `make reset`.
- [x] I confirmed that there are no unintended functional regressions in this branch:
  - [x] I have managed to pass the onboarding flow
  - [x] Objects and Findings are created properly
  - [x] Tasks are created and completed properly
- [x] I confirmed that the PR's advertised `feature` or `hotfix` works as intended.
- [x] I checked the logs for errors and/or warnings and made issues where necessary

### What works:
Looks good. Both creating and editing events are shown up and the boefje variant runs.

### What doesn't work:
n/a

### Bug or feature?:
n/a

**Review by underdarknl — APPROVED** [review]

**Pull request merged**